src/state/GlobalStore.js

```diff
@@ -43,6 +41,10 @@ class GlobalStore {
 
   addReducers() {
     addReducer('empty', () => {
+      resetGlobal();
```

Comment:
the global callback that updates local storage was not receiving the reset global state, perhaps the async operations were occurring in a different order than expected. `resetGlobal()` completely clears the state, then callbacks and reducers are re-added and the fresh global state is returned to repopulate the state. This triggers the correct order of saving the state to local storage after it has been cleared.